### DISCUSSION TITLE
Starlight docs sidebar ability to "lock" entries in the list?

### DISCUSSION THREAD

**Discussion by lirantal** [withastro/starlight#2652] [Q&A]
I want to build on the Starlight docs and have the sidebar that shows the chapters take a predicate to change the behavior of whether to show the item in the list or show "Locked" text instead. This is in order to satisfy so that some docs pages are public to access and others are enabled only after authentication.

**Comment by lirantal** [comment]
Update: I found out that Starlight has its own plugins that extend it just like Astro (woohoo!), then I found out there's a sidebar navigation plugin (!) and then that lead me to realize I can override Starlight components, including its own built-in Sidebar components [like this plugin is doing](https://github.com/lorenzolewis/starlight-utils/blob/main/packages/starlight-utils/components/Sidebar.astro).

I still need to spend more time hacking on it but it looks like I can define my own Sidebar and that will be server-side rendered, in which I can make server calls to determine the behavior and contents of the Sidebar links. Will share an update later.

**Comment by HiDeoo** [comment]
Thanks for the feedback.

Looks like you found quite a few resources on this topic, and I'd like to add a few:

- An [example](https://github.com/HiDeoo/starlight-better-auth-example) of project using the Better Auth library to add authentication to a Starlight documentation site. Users can create an account, sign in, and sign out. Some pages are protected and require the user to be authenticated to access them or even be visible in the documentation sidebar.
- The [route data](https://starlight.astro.build/guides/route-data/) documentation which is an object containing all the information required to render a single page. It includes information for the current page, including the sidebar, which can be [customized](https://starlight.astro.build/guides/route-data/#customizing-route-data).
